### ISSUE TITLE
Scrolling the 'All Applications' menu is slow (and causes high CPU load)

### ISSUE THREAD

**Issue by crahan** [linuxmint/cinnamon#7191]
```
 * Cinnamon 3.6.7
 * Mint 18.3
 * Intel 2nd Generation Core Processor Family Integrated Graphics Controller (X11 driver: intel)
 * 64 bit
 ```

**Issue**
When hovering the mouse over an application entry in the `All Applications` category of the Cinnamon menu and scrolling with 2 fingers on a trackpad or scrolling using a trackball, the scroll animation is choppy and results in high CPU load (100%).

However, when the mouse pointer hovers over the scrollbar for the `All Applications` category the scrolling is smooth and CPU load is much lower (half). The only difference is the app title and description lookup that appears near the bottom of the Cinnamon menu. When the title and description is showing scrolling is slow. When the information does not show scrolling is smooth.

Also, this only happens in the `All Applications` category and not with any of the other application categories (possibly due to the amount of entries?).

**Steps to reproduce**
* Open Cinnamon menu
* Go to the `All Applications` category
* Hover the mouse over `Account details`
* Scroll using 2 fingers on a trackpad or using a trackball device
* Scroll animation is choppy and skips over entries

**Expected behaviour**
Scrolling up and down the `All Application` category should be smooth.

**Other information**
None

**Comment by xenopeek** [comment]
I could reproduce the high CPU load using scrollwheel on a regular mouse.

- Go to the All Applications category in the menu and use the scrollwheel while hovering the mouse cursor over the list of applications. In System Monitor I see CPU load spike to 100%.
- If I instead hover the mouse cursor over the scrollbar and then use the scrollwheel, CPU load is around 50%.

When the mouse cursor is hovering over the list of applications, as you scroll, the name and description of each application you scroll over is displayed at the bottom of the menu. When hovering over the scrollbar that doesn't happen. Could this looking up of the application name and description cause the slow down?

**Comment by pyroxar** [comment]
I confirm this bug too on my laptop with Linux Mint 18.3.
![cpu](https://user-images.githubusercontent.com/23406555/34915769-d776f760-f92b-11e7-8ebe-f5809e56e1e9.png)

**Comment by bambamal** [comment]
+1 I noticed the same issue. Scrolling with a touchpad or a mousewheel is very slow and sluggish. Scrolling with the scrollbar works just fine.

**Comment by Bodai-76** [comment]
Hello guys.
I have the same symptoms as described by craham. When scrolling through the "All Programs" category of the menu, I get a very bad performance of the menu and the scroll seems to be VERY choppy. It can be reproduced with the arrows keys with a keyboard as well, not only when scrolling with the touchpad.

It seems like when there is a program selected on the menu listing on the "All Programs" category, the menu needs to look for something and I get the load spikes on the processor and the scroll is really choppy. If you choose the all programs category and scroll the programs with the up or down arrow of the keyboard, the problem is still there. If you select any other category, the scrolling works fine.

**When the problem occur?**
When there is a program hoovered by the mouse or selected with a keyboard on the **_All _Programs__**  category (when the menu shows the name of the program selected and it's description on the bottom of the menu). If you scroll with the mouse over the scroll bar on the right side of the menu, there is no problem. 

If needed, I can try to record a video of the problem to show it better.

**Comment by chaoticsoap** [comment]
Manjaro Cinnamon here, this problem is still active on both of my installs, laptop and tower.

**Comment by gronostajo** [comment]
I'm affected too. I'm trying to use Cinnamon on a pretty low-spec netbook. Touchpad scrolling when pointer is over the scrollbar is smooth, but when I hover over the list, it results in a two-second freeze. Maybe drawing item's mouseover background is CPU-intensive and scrolling causes multiple redraws?

**Comment by eyes-0nly** [comment]
I can confirm that the issue is still present on Cinnamon 4.0.2

**Comment by Nesty92** [comment]
Any updates ? this issue is still present

**Comment by Vahan86** [comment]
@crahan, is this still an issue for you with Cinnamon 4.4?

**Comment by pyroxar** [comment]
I can't reproduce this bug.
![obraz](https://user-images.githubusercontent.com/23406555/74608264-e51c7d00-50df-11ea-9502-1a1f16d7017e.png)

**Comment by pyroxar** [comment]
I can't reproduce this bug.
![obraz](https://user-images.githubusercontent.com/23406555/74608272-eea5e500-50df-11ea-9405-3f355d56a9bf.png)

**Comment by Vahan86** [comment]
@icarter09, @xenopeek, can we close this issue?

**Comment by crahan** [comment]
I can confirm that this has been resolved for me now (19.3 and 20).
